internal/services/producer.go

```diff
@@ -18,16 +19,24 @@ import (
 )
 
 var (
-	recordsProduced = promauto.NewCounter(prometheus.CounterOpts{
+	recordsProduced = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name:      "records_produced_total",
 		Namespace: "strimzi_canary",
 		Help:      "The total number of records produced",
-	})
-	recordsProducedFailed = promauto.NewCounter(prometheus.CounterOpts{
+	}, []string{"clientid", "partition"})
+
+	recordsProducedFailed = promauto.NewCounterVec(prometheus.CounterOpts{
 		Name:      "records_produced_failed_total",
 		Namespace: "strimzi_canary",
 		Help:      "The total number of records failed to produce",
-	})
+	}, []string{"clientid", "partition"})
+
+	recordsProducedLatency = promauto.NewHistogramVec(prometheus.HistogramOpts{
+		Name:      "records_produced_latency",
+		Namespace: "strimzi_canary",
+		Help:      "Records end-to-end latency in milliseconds",
+		Buckets:   []float64{100, 200, 400, 800, 1600},
```

Comment:
Might want to make this configurable. Even if not, 100ms wouldn't really give any resolution at the lower end, so it will take a little experimentation to determine the ideal range.